packages/core/config.js

```diff
@@ -8,52 +14,60 @@ module.exports.schema = {
     validate: stringWithLength
   },
   appVersion: {
-    defaultValue: () => null,
+    defaultValue: () => undefined,
     message: 'should be a string',
-    validate: value => value === null || stringWithLength(value)
+    validate: value => value === undefined || stringWithLength(value)
   },
   appType: {
-    defaultValue: () => null,
+    defaultValue: () => undefined,
     message: 'should be a string',
-    validate: value => value === null || stringWithLength(value)
+    validate: value => value === undefined || stringWithLength(value)
   },
-  autoNotify: {
-    defaultValue: () => true,
+  autoDetectErrors: {
+    defaultValue: () => undefined,
     message: 'should be true|false',
-    validate: value => value === true || value === false
+    validate: (value) => value === true || value === false || value === undefined
   },
-  beforeSend: {
-    defaultValue: () => [],
-    message: 'should be a function or array of functions',
-    validate: value => typeof value === 'function' || (isArray(value) && filter(value, f => typeof f === 'function').length === value.length)
+  autoDetectUnhandledRejections: {
+    defaultValue: () => undefined,
+    message: 'should be true|false',
+    validate: (value) => value === true || value === false || value === undefined
+  },
+  context: {
+    defaultValue: () => undefined,
+    message: 'should be a string',
+    validate: value => value === undefined || stringWithLength(value)
   },
+  onError: listOfCallbacks,
+  onBreadcrumb: listOfCallbacks,
+  onSession: listOfCallbacks,
   endpoints: {
     defaultValue: () => ({
       notify: 'https://notify.bugsnag.com',
       sessions: 'https://sessions.bugsnag.com'
     }),
-    message: 'should be an object containing endpoint URLs { notify, sessions }. sessions is optional if autoCaptureSessions=false',
+    message: 'should be an object containing endpoint URLs { notify, sessions }. sessions is optional if autoTrackSessions=false',
     validate: (val, obj) =>
       // first, ensure it's an object
       (val && typeof val === 'object') &&
       (
         // endpoints.notify must always be set
         stringWithLength(val.notify) &&
         // endpoints.sessions must be set unless session tracking is explicitly off
-        (obj.autoCaptureSessions === false || stringWithLength(val.sessions))
+        (obj.autoTrackSessions === false || stringWithLength(val.sessions))
       ) &&
       // ensure no keys other than notify/session are set on endpoints object
-      filter(keys(val), k => !includes([ 'notify', 'sessions' ], k)).length === 0
+      filter(keys(val), k => !includes(['notify', 'sessions'], k)).length === 0
   },
-  autoCaptureSessions: {
+  autoTrackSessions: {
     defaultValue: (val, opts) => opts.endpoints === undefined || (!!opts.endpoints && !!opts.endpoints.sessions),
```

Comment:
The logic here makes it `true` by default, but only if a session endpoint is supplied. It's not doing any URL validation.

The use case for this is a little niche, I guess, but it would be for an on-prem user who doesn't want to supply a session endpoint because they don't want sessions. Is that not supported on Android (just supplying a notify endpoint)? If not I guess we could eliminate that possibility here too.

This diff shows how the simplest way of disabling session tracking changes if we do that: 

```diff
  {
   endpoints: {
     notify: "https://notify.my-bugsn.ag",
+    sessions: "https://sessions.my-bugsn.ag"
   },
+  autoTrackSessions: false
}
```